esp_idf_monitor/base/serial_reader.py

```diff
@@ -102,30 +102,38 @@ def run(self):
     def close_serial(self):
         if sys.platform == 'linux':
             # Avoid waiting for 30 seconds before closing the serial connection
-            self.set_closing_wait(delay_sec=1)
+            self._disable_closing_wait_or_discard_data()
         self.serial.close()
 
-    def set_closing_wait(self, delay_sec):  # type: (int) -> None
+    def _disable_closing_wait_or_discard_data(self):  # type: () -> None
         import fcntl
         import struct
         import termios
 
         # `serial_struct` format based on linux kernel source:
         # https://github.com/torvalds/linux/blob/25aa0bebba72b318e71fe205bfd1236550cc9534/include/uapi/linux/serial.h#L19
         struct_format = 'iiIiiiiiHcciHHPHIL'
-        buf = bytes(struct.calcsize(struct_format))
+        buf = bytearray(struct.calcsize(struct_format))
 
         # get serial_struct
         fcntl.ioctl(self.serial.fd, termios.TIOCGSERIAL, buf)
```

Comment:
This should resolve your issue with bytes (I am sorry it was caused by me), but bytearray is okay as well, if you fix the pre-commit check.
```suggestion
        buf = fcntl.ioctl(self.serial.fd, termios.TIOCGSERIAL, buf)
```